crates/accelerate/src/target_transpiler/mod.rs

```diff
@@ -0,0 +1,1238 @@
+    /// (for multi-qubit operations).
+    ///
+    /// Args:
+    ///     strict_direction (bool): If set to ``True`` the multi-qubit
+    ///         operations considered as non-global respect the strict
+    ///         direction (or order of qubits in the qargs is significant). For
+    ///         example, if ``cx`` is defined on ``(0, 1)`` and ``ecr`` is
+    ///         defined over ``(1, 0)`` by default neither would be considered
+    ///         non-global, but if ``strict_direction`` is set ``True`` both
+    ///         ``cx`` and ``ecr`` would be returned.
+    ///
+    /// Returns:
+    ///     List[str]: A list of operation names for operations that aren't global in this target
+    #[pyo3(signature = (/, strict_direction=false,), text_signature = "(/, strict_direction=False)")]
+    pub fn get_non_global_operation_names(
+        &mut self,
+        py: Python<'_>,
+        strict_direction: bool,
+    ) -> PyObject {
+        self.get_non_global_op_names(strict_direction).to_object(py)
+    }
+
+    // Class properties
+
+    /// The set of qargs in the target.
+    #[getter]
+    fn qargs(&self) -> Option<Vec<Option<Qargs>>> {
+        self.get_qargs()
+            .map(|qargs| qargs.into_iter().cloned().collect())
+    }
+
+    /// Get the list of tuples ``(:class:`~qiskit.circuit.Instruction`, (qargs))``
+    /// for the target
+    ///
+    /// For globally defined variable width operations the tuple will be of the form
+    /// ``(class, None)`` where class is the actual operation class that
+    /// is globally defined.
+    #[getter]
+    pub fn instructions(&self, py: Python<'_>) -> PyResult<Vec<(PyObject, Option<Qargs>)>> {
+        // Get list of instructions.
+        let mut instruction_list: Vec<(PyObject, Option<Qargs>)> = vec![];
+        // Add all operations and dehash qargs.
+        for (op, props_map) in self.gate_map.iter() {
+            for qarg in props_map.keys() {
+                let instruction_pair = (self._gate_name_map[op].object.clone_ref(py), qarg.clone());
+                instruction_list.push(instruction_pair);
+            }
+        }
+        // Return results.
+        Ok(instruction_list)
+    }
+    /// Get the operation names in the target.
+    #[getter]
+    pub fn operation_names(&self) -> Vec<String> {
+        self.gate_map.keys().cloned().collect()
+    }
+
+    /// Get the operation objects in the target.
+    #[getter]
+    pub fn operations(&self) -> Vec<&PyObject> {
+        return Vec::from_iter(self._gate_name_map.values().map(|x| &x.object));
+    }
+
+    /// Returns a sorted list of physical qubits.
+    #[getter]
+    pub fn physical_qubits(&self) -> Vec<usize> {
+        Vec::from_iter(0..self.num_qubits.unwrap_or_default())
+    }
+
+    // Magic methods:
+
+    fn __len__(&self) -> PyResult<usize> {
+        Ok(self.gate_map.len())
+    }
+
+    fn __getstate__(&self, py: Python<'_>) -> PyResult<Py<PyList>> {
+        let result_list = PyList::empty_bound(py);
```

Comment:
Since there are so many fields here I'd probably use a dict, so that you can access by name. With 15 elements tracking which index maps to which field seems quite error prone when we need to update this.